crates/wash-lib/src/build/component.rs

```diff
@@ -118,6 +101,28 @@ pub fn build_component(
     }
 }
 
+pub fn adapt_component_to_wasi_preview2(
+    component_wasm_path: &PathBuf,
```

Comment:
```suggestion
    component_wasm_path: impl AsRef<Path>
```

Also, what do you think about this being `pub` versus `pub(crate)`? I think the function is definitely useful as `pub`, but I wonder if it would be easier to use if it didn't require a `&ComponentConfig`, but exposed the things it actually uses -- the path to the WASM, etc.

As far as I can tell the vast majority of `ComponentConfig` is not used here.

That said, this is absolutely a NIT -- I'm just as happy to mark it `pub(crate)` and not expose something new to the outside world (which is still mostly us/`wash-cli`)

Reply:
made it `pub(crate)` for now and addressed argument trait 👍